### PR TITLE
Experimental http wip

### PR DESCRIPTION
**Pull request by simongdavies**
Adds support for experimental http #4

### CODE REVIEW

File: source/wagi/WasiExperimentalHttp/HttpRequestHandler.cs [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+  using Wasmtime;
+
+  /// <summary>
+  /// HttpRequestHandler provides support for wasi_experimental_http.
+  /// </summary>
+  internal class HttpRequestHandler : IDisposable
+  {
+    private const string ModuleName = "wasi_experimental_http";
+    private const string MemoryName = "memory";
+    private const int MaxResponses = 10;
```
Comment (radu-matei) [inline source/wagi/WasiExperimentalHttp/HttpRequestHandler.cs:23]: The default behavior in the Rust implementation is that if the user does not set a maximum here, it is only limited by the underlying platform (i.e. when the host runs out of outbound connections, for example).
Do we want to enforce a number here?

Comment (simongdavies) [reply source/wagi/WasiExperimentalHttp/HttpRequestHandler.cs:23]: OK, I will make it configurable via config settings and default it to a higher value.

**Comment by radu-matei** [comment]
Tried running the `simplehttp` example:

```
➜ curl -X POST -d "sddf" localhost:8888/test
System.InvalidOperationException: Invalid non-ASCII or control character in header: 0x0000
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpHeaders.ThrowInvalidHeaderCharacter(Char ch)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpHeaders.ValidateHeaderValueCharacters(StringValues headerValues)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseHeaders.AddValueFast(String key, StringValues value)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpHeaders.System.Collections.Generic.IDictionary<System.String,Microsoft.Extensions.Primitives.StringValues>.Add(String key, StringValues value)
   at System.Collections.Generic.CollectionExtensions.TryAdd[TKey,TValue](IDictionary`2 dictionary, TKey key, TValue value)
   at Deislabs.WAGI.Helpers.WAGIHost.AddHeader(String key, String value) in D:\projects\deislabs\wagi-dotnet\source\wagi\Helpers\WAGIHost.cs:line 328
   at Deislabs.WAGI.Helpers.WAGIHost.<>c__DisplayClass21_0.<ProcessOutput>b__0(String header) in D:\projects\deislabs\wagi-dotnet\source\wagi\Helpers\WAGIHost.cs:line 272
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Deislabs.WAGI.Helpers.WAGIHost.ProcessOutput(String outputPath) in D:\projects\deislabs\wagi-dotnet\source\wagi\Helpers\WAGIHost.cs:line 262
   at Deislabs.WAGI.Helpers.WAGIHost.ProcessRequest() in D:\projects\deislabs\wagi-dotnet\source\wagi\Helpers\WAGIHost.cs:line 123
   at Deislabs.WAGI.Extensions.HttpContextExtensions.RunWAGIRequest(HttpContext context, String wasmFile, IHttpClientFactory httpClientFactory, String entryPoint, String moduleType, IDictionary`2 volumes, IDictionary`2 environment, List`1 allowedHosts, Int32 maxHttpRequests) in D:\projects\deislabs\wagi-dotnet\source\wagi\Extensions\HttpContextExtensions.cs:line 32
   at Deislabs.WAGI.Extensions.EndpointRouteBuilderExtensions.<>c__DisplayClass0_1.<<MapWASMModules>b__0>d.MoveNext() in D:\projects\deislabs\wagi-dotnet\source\wagi\Extensions\EndpointRouteBuilderExtensions.cs:line 118
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

HEADERS
=======
Content-Type: application/x-www-form-urlencoded
Accept: */*
Host: localhost:8888
User-Agent: curl/7.55.1
Content-Length: 4
```

**Comment by simongdavies** [comment]
@radu-matei was an issue with new lines in as sample on windows, updated.

**Review by radu-matei — APPROVED** [review]
Manually tested with a Rust module as well (and on Windows), LGTM.

**Pull request merged**